inc/error.hpp

```diff
@@ -19,6 +19,12 @@
 // HTTP REQUEST HEADER ERRORS
 #define ERR_HEADER_COLON_WHITESPACE "Invalid HTTP request: Whitespace between header field-name and colon detected"
 #define ERR_HEADER_NAME_INVALID_CHAR "Invalid HTTP request: Invalid char in header field name"
+#define ERR_MISSING_HOST_HEADER "Invalid HTTP request: Missing host header"
+#define ERR_EMPTY_HOST_VALUE "Invalid HTTP request: Missing hostname"
+#define ERR_MULTIPLE_HOST_HEADERS "Invalid HTTP request: Multiple host headers"
+#define ERR_INVALID_HOSTNAME "Invalid HTTP request: Invalid hostname"
+#define ERR_INVALID_IP "Invalid HTTP request: Invalid IP"
+#define ERR_INVALID_IP_WITH_PORT "Invalid HTTP request: Invalid IP with port"
```

Comment:
I would only use the first error message as it bring the point accross.

If you want to keep all three I would at least specify that the invalid IP (with port) was determined in context of hostname

Reply:
It mostly helped me with debugging. Specified to hostname context.